pkg/validator/validator.go

```diff
@@ -1,25 +1,65 @@
 package validator
 
 import (
-	"fmt"
-
 	y "gopkg.in/yaml.v3"
 )
 
 type Error struct {
-	Msg string
+	Details []error
+	Msg     string
 }
 
 func (e Error) Error() string {
 	return e.Msg
 }
 
-func NewError(msg string) error {
-	return Error{Msg: msg}
+func (e Error) Unwrap() []error {
+	return e.Details
+}
+
+func NewError(msg string) Error {
```

Comment:
It kinda feels weird we're calling NewError which may not be an error, but a grouping of errors which we check the length of later in ErrOrNil. Would renaming the struct to Validator or similar help?